src/pykeen/nn/functional.py

```diff
@@ -1107,17 +1109,18 @@ def p_rotate_interaction(
     h: torch.FloatTensor,
     r: torch.FloatTensor,
     t: torch.FloatTensor,
-    **kwargs,
+    p: Union[str, int, float] = 2,
+    power_norm: bool = False,
 ) -> torch.FloatTensor:
     r"""
-    Evaluate the pRotatE scoring function.
+    Evaluate the pRotatE scoring function from [sun2019]_.
 
     The score function is given as
 
     .. math ::
         \|\sin ((h + r - t)/2)\|_p
     """
-    return negative_norm((0.5 * (h + r - t)).sin(), **kwargs)
+    return negative_norm((0.5 * (h + r - t)).sin(), p=p, power_norm=power_norm)
```

Comment:
@cthoyt I did not yet verify that this is indeed the variant from Sun, but rather took this from the HAKE paper, where the authors claim this is the pRotate variant from Sun.

Reply:
https://github.com/DeepGraphLearning/KnowledgeGraphEmbedding/blob/2e440e0f9c687314d5ff67ead68ce985dc446e3a/codes/model.py#L230-L248

`score = norm(sin(h + r - t)) * modulus`, where modulus is a global parameter. So it is not exactly the same as the Sun variant.

Reply:
then let's provide both possible implementations. The trick will be to pick names that are not confusing

Reply:
Your version can be BRotatE ;)

Reply:
So can we parametrize the 0.5 and also an outer multipler then have a single general interaction?